rivescript/parser.py

```diff
@@ -574,6 +574,14 @@ def check_syntax(self, cmd, line):
                     angle += 1
                 elif char == '>':
                     angle -= 1
+                elif char == '|':
+                    if parens == 0 and square == 0:   # Pipe outside the alternative and option
+                        return "Pipe | must be within parenthesis brackets or square brackets"
+
+                total = parens + square + curly + angle   # At each character, not more than 1 bracket opens
+                for special_char_count in [parens, square, curly, angle, total]:
+                    if special_char_count not in (0, 1):
+                        return "Unbalanced brackets"
```

Comment:
Yeah the only curly bracket tag allowed in triggers is `{weight}`, and it doesn't make sense to embed anything inside it (the regexp that parses the `{weight}` tag only expects to find a number in there).

So your idea sounds good -- while doing things like `((hello|hi) (robot|computer))` are allowed in regexps, I don't condone that kind of syntax in RiveScript and none of the docs mention it, so I _would_ like it to become a syntax error. :smile: